File: CSharp/Services/ComponentBService.cs
namespace CSharpDi.Services
{
    public class ComponentBService
    {
        private readonly Logger logger;
        private readonly SingletonState singletonState;
        private readonly TransientState transientState;

        public ComponentBService(Logger logger, SingletonState singletonState, TransientState transientState)
        {
            this.logger = logger;
            this.singletonState = singletonState;
            this.transientState = transientState;
        }

        public void DoIt()
        {
            logger.LogInformation($"ComponentAService (SingletonState) {singletonState.GetValue()}");
            logger.LogInformation($"ComponentAService (TransientState) {transientState.GetValue()}");
        }
    }
}

File: TypeScript/services/ComponentBService.ts
import { injectable } from "tsyringe";
import { Logger } from "./Logger";
import { SingletonState } from "./SingletonState";
import { TransientState } from "./TransientState";

@injectable()
export class ComponentBService {
    constructor(private logger: Logger, private singletonState: SingletonState, private transientState: TransientState) {}

    public DoIt() {
        this.logger.LogInformation(`ComponentBService (SingletonState) ${this.singletonState.getValue()}`)
        this.logger.LogInformation(`ComponentBService (TransientState) ${this.transientState.getValue()}`)
    }
}